README.md

```diff
@@ -1,133 +1,616 @@
-## Smithy Typescript
+# <img alt="Smithy" src="https://github.com/smithy-lang/smithy/blob/main/docs/_static/favicon.png?raw=true" width="28"> Smithy TypeScript
 
-Smithy code generators for TypeScript.
+> **WARNING: Smithy TypeScript is currently in [Developer Preview](https://aws.amazon.com/blogs/devops/smithy-server-and-client-generator-for-typescript). All interfaces and supported JavaScript platforms are subject to change.**
+> 
+> **NOTE: For Node.js only, currently versions >= 14 are supported, and are subject to change.**
 
-**WARNING: All interfaces are subject to change.**
+`smithy-typescript` includes the reference implementations of the [Smithy](https://smithy.io/) code generators for [TypeScript](https://www.typescriptlang.org/).
```

Comment:
Given there are now quite a few sections it might be nice to add a small table of contents.

Reply:
GitHub has a built-in table of contents for markdown files, so I think this would be redundant.